### DISCUSSION TITLE
How to computed properties

### DISCUSSION THREAD

**Discussion by matzeeable** [TanStack/query#3044] [Q&A] [answered]
Hey !

React Query looks so promising. I just read the complete documentation and tried to cover all cases within your existing application with MobX.

I found one thing I could not find a solution: `mobx.computed`: https://mobx.js.org/computeds.html

How could this be covered by RQ?

**Comment by TkDodo** [comment, marked as answer]
`select` is probably the closest thing you are looking for: https://tkdodo.eu/blog/react-query-data-transformations#3-using-the-select-option

**Reply by matzeeable** [reply]
Thanks for your fast answer. As I can see from the documentation, the result is not memorized. Imagine, you have a complex and "long-running" compution, this is done for every component you are using the query, nor?

**Reply by TkDodo** [reply]
the result is memoized and structurally shared _per observer_, if you provide a stable function. If it has no dependencies, you can just extract it to a stable reference. If it has dependencies, `useCallback`.

if you need to store the result for multiple observers, you can combine it with something like [reselect](https://github.com/reduxjs/reselect) to memoize per input.

**Reply by matzeeable** [reply]
Awesome! Thanks!
